src/Database/LSMTree/Internal/Run/Index/Compact.hs

```diff
@@ -698,7 +698,7 @@ fromSBS (SBS ba') = do
     let ba = ByteArray ba'
     let len8 = sizeofByteArray ba
     when (mod8 len8 /= 0) $ Left "Length is not multiple of 64 bit"
-    when (len8 < 36) $ Left "Doesn't contain header and footer"
+    when (len8 < 32) $ Left "Doesn't contain header and footer"
```

Comment:
Should it be `28`? It's 1 `Word32` in the header, and 3 `Word64`s in the footer

Reply:
Due to the 64 bit alignment constraint we need at least 4 Word64 (but of course it doesn't really make a difference since we enforce `mod8 len8 == 0` above).

Reply:
Are you both happy with this then?

Reply:
> Due to the 64 bit alignment constraint we need at least 4 Word64 (but of course it doesn't really make a difference since we enforce `mod8 len8 == 0` above).

Good point, thanks 

Different Q: technically, if we consider that the range finder array contains 2 elements at least, which adds at least 8 bytes, and there is the clash map size, which adds 8 bytes, the minimum size for a sensible header/footer should be even higher than `36`?

Reply:
Yes, we could calculate a slightly higher minimum size. However, the checks for the other components are still needed either way, so only the error message would change in some cases.
In terms of simplicity of the code, I think it makes sense for this check to not get ahead of itself and only make sure we can safely read the header and footer.